### PR TITLE
fix dev code for webpack compatibility

### PR DESCRIPTION
**Pull request by cherifGsoul**
This fixes the first part of this issue canjs/canjs#4170 to remove debug code from the production builds.

### CODE REVIEW

**Comment by cherifGsoul** [comment]
@matthewp @justinbmeyer is this can be merged?

File: can-define.js [hunk 1]
```diff
@@ -290,13 +300,15 @@ define.property = function(typePrototype, prop, definition, dataInitializers, co
...
 
 		//!steal-remove-start
-		// If value is an object or array, give a warning
-		if (definition.default !== null && typeof definition.default === 'object') {
-			canLogDev.warn("can-define: The default value for " + prop + " is set to an object. This will be shared by all instances of the DefineMap. Use a function that returns the object instead.");
-		}
-		// If value is a constructor, give a warning
-		if (definition.default && canReflect.isConstructorLike(definition.default)) {
-			canLogDev.warn("can-define: The \"default\" for " + prop + " is set to a constructor. Did you mean \"Default\" instead?");
+			if(process.env.NODE_ENV !== 'production') {
```
Comment (justinbmeyer) [inline can-define.js]: I think there's an indentation problem here.

File: list/list.js [hunk 2]
```diff
@@ -590,44 +599,46 @@ var DefineList = Construct.extend("DefineList",
 		// pass `key` to only log the matching event, e.g: `list.log("add")`
 		log: function(key) {
 			//!steal-remove-start
-			var instance = this;
+			if(process.env.NODE_ENV !== 'production') {
```
Comment (justinbmeyer) [inline list/list.js]: Might it have been easier to have log be a no-op here and at the bottom of this file:

```js
//!steal-remove...
if(process.env...){
  DefineList.prototype.log = function(){ ... }
}
//!steal-remove..
```

File: map/map.js [hunk 3]
```diff
@@ -433,37 +435,39 @@ var DefineMap = Construct.extend("DefineMap",{
 	// pass `key` to only log the matching property, e.g: `map.log("foo")`
 	log: function(key) {
 		//!steal-remove-start
-		var instance = this;
-
-		var quoteString = function quoteString(x) {
-			return typeof x === "string" ? JSON.stringify(x) : x;
-		};
+		if(process.env.NODE_ENV !== 'production') {
```
Comment (justinbmeyer) [inline map/map.js]: same comment about log from above.  Btw, could `log` be shared?

Comment (cherifGsoul) [reply map/map.js]: Even if they don't share the same event types?

Comment (cherifGsoul) [reply map/map.js]: The log is shared now.

**Comment by cherifGsoul** [comment]
@justinbmeyer ```log``` function is in helpers now

**Comment by justinbmeyer** [comment]
the ensure_meta should probably be put in define_helpers too.  I think we should avoid creating modules unnecessarily. It adds to the runtime.

**Comment by cherifGsoul** [comment]
@justinbmeyer Can we handle ensure_meta in other PR?

**Comment by justinbmeyer** [comment]
Yes. We can do it later.

**Pull request merged**